From ffd906c1a60b6fd3ea83983ac8b382f103df43cd Mon Sep 17 00:00:00 2001
From: Jaikrishna Nemallapudi <nemallapudi.jaikrishna@intel.com>
Date: Thu, 28 May 2020 23:17:14 +0530
Subject: [PATCH] input: passthrough virtio-input device

The virtio-input device event should be passthrough
and sent to uplayer directly, since the events get
from host, and already handled in host input driver.

Tracked-On: OAM-91172
Signed-off-by: Jaikrishna Nemallapudi <nemallapudi.jaikrishna@intel.com>
Signed-off-by: Yang, Dong <dong.yang@intel.com>
---
 drivers/input/input.c | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/drivers/input/input.c b/drivers/input/input.c
index 95e5dac48164..39fc4fe18c8b 100644
--- a/drivers/input/input.c
+++ b/drivers/input/input.c
@@ -377,6 +377,23 @@ static void input_handle_event(struct input_dev *dev,
 		return;
 
 	disposition = input_get_disposition(dev, type, code, &value);
+	if (dev->phys && !strncmp(dev->phys, "virtio", 6)) {
+		struct input_value *v = NULL;
+		if (!dev->vals)
+			return;
+
+		v = &dev->vals[dev->num_vals++];
+		if (!v)
+			return;
+
+		v->type = type;
+		v->code = code;
+		v->value = value;
+		input_pass_values(dev, dev->vals, dev->num_vals);
+		dev->num_vals = 0;
+		return;
+	}
+
 	if (disposition != INPUT_IGNORE_EVENT && type != EV_SYN)
 		add_input_randomness(type, code, value);
 
-- 
2.24.0

